### ISSUE TITLE
CORS header support

### ISSUE THREAD

**Issue by developernaren** [thephpleague/oauth2-server#359]
I am trying to create an oauth resource server and I am using Lumen. The request come in from a different domain, So I should allow all the request to come in.
I have used a Middleware

    class CorsMiddleware {

    public function handle($request, \Closure $next) {
        $response = $next($request);

        $response->header('Access-Control-Allow-Methods', 'HEAD, GET, POST, PUT, PATCH, DELETE, OPTIONS');
        $response->header('Access-Control-Allow-Headers', $request->header('Access-Control-Request-Headers'));
        $response->header('Access-Control-Allow-Origin', '*');

        return $response;
    }

    }
which attached headers to all the response. This is working fine except for when the username/ password is invalid. Lumen does not hit the this method, instead I see that it throws an exception. I tried to catch it with this

    $app->error(function(League\OAuth2\Server\Exception\OAuthException $exception)
     {
    $response = Response::make([
        'error'   => $exception->errorType,
        'message' => $exception->getMessage()
    ], 401);

        $response->header('Access-Control-Allow-Origin' , '*');
    $response->header('Access-Control-Allow-Methods', 'GET, POST, PUT, OPTIONS');
    $response->header('Access-Control-Allow-Headers', 'Origin, Content-Type, Accept, Authorization');

    return $response;
    });

a solution suggested by [@JoeyHoutenbos](https://github.com/JoeyHoutenbos)

But none of these seem to work and this method seems to override all this and the header is sent empty because of this method 

     public function getHttpHeaders()
        {

        $headers = [];        
        switch ($this->httpStatusCode) {
            case 401:
                $headers[] = 'HTTP/1.1 401 Unauthorized';
                break;
            case 500:
                $headers[] = 'HTTP/1.1 500 Internal Server Error';
                break;
            case 501:
                $headers[] = 'HTTP/1.1 501 Not Implemented';
                break;
            case 400:
            default:
                $headers[] = 'HTTP/1.1 400 Bad Request';
                break;
            }

`League\OAuth2\Server\Exception\OAuthException;`

I literally get the status code in my `headers 0 index` in my angular app

to fix this I changed the code to 

    $headers = [];        
            $headers ['Access-Control-Allow-Methods'] = 'HEAD, GET, POST, PUT, PATCH, DELETE,        OPTIONS';
                $headers ['Access-Control-Allow-Origin'] = '*';

and it is working like a charm.

Is there a way that this can be fixed without messing with the vendor file?

**Comment by HonzaMac** [comment]
Any progress on that? This is little bit show stopper for this oauth implementation

**Comment by developernaren** [comment]
I am stuck with this implementation and there is no answer to the question. Would you like me to help and make a pull request for this error? Should I see if I can help?

**Comment by alexbilbie** [comment]
Hi @developernaren, sorry for the late reply I've been away on holiday.

Both the [resource server example](https://github.com/thephpleague/oauth2-server/blob/master/examples/relational/api.php#L105) and the [auth server example](https://github.com/thephpleague/oauth2-server/blob/master/examples/relational/authcode_grant.php#L90) have examples of setting additional headers on the response object.

I'm not familiar with Lumen, but why don't these examples work for you?

**Comment by developernaren** [comment]
I hope you had a wonderful holiday. 

I actually created (this issue)[https://github.com/lucadegasperi/oauth2-server-laravel/issues/466] . then (Luca)[https://github.com/lucadegasperi] suggested that I create issue here. I had not looked into the code after I created the issue hoping that I will get reply but I remembered the issue today while looking into something else and looked around and found the solution and I suggested the same to Luca in the same thread. I think it is simpler to make the edit there.Its a coincidence that you replied the same day that I suggested the solution and it seems that your example is almost exactly as the solution that I suggested. Great Minds, I guess ;)

I am waiting for Luca's reply now.
